core/libs/shared/src/path_ops.rs

```diff
@@ -15,69 +15,8 @@ use crate::{validate, SharedError, SharedResult};
 impl<Base, Local> LazyStaged1<Base, Local>
 where
     Base: TreeLike<F = SignedFile>,
-    Local: TreeLikeMut<F = Base::F>,
+    Local: TreeLike<F = Base::F>,
 {
-    pub fn create_link_at_path(
-        &mut self, path: &str, target_id: Uuid, root: &Uuid, account: &Account, pub_key: &PublicKey,
-    ) -> SharedResult<Uuid> {
-        validate::path(path)?;
-        let file_type = FileType::Link { target: target_id };
-        let path_components = split_path(path);
-        self.create_at_path_helper(file_type, path_components, root, account, pub_key)
-    }
-
-    pub fn create_at_path(
-        &mut self, path: &str, root: &Uuid, account: &Account, pub_key: &PublicKey,
-    ) -> SharedResult<Uuid> {
-        validate::path(path)?;
-        let file_type = if path.ends_with('/') { FileType::Folder } else { FileType::Document };
-        let path_components = split_path(path);
-        self.create_at_path_helper(file_type, path_components, root, account, pub_key)
-    }
```

Comment:
these and similar functions moved to an `impl` block which requires `Local` to be `TreeLikeMut` (this impl block modified to accept a `Local` that's only `TreeLike`)